src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -208,6 +220,49 @@ private BidRequest createSingleRequest(Imp imp, BidRequest bidRequest, boolean u
                 .build();
     }
 
+    private String makeUri(BidRequest bidRequest) {
+        final String tkXint = tkXintValue(bidRequest);
+        if (!StringUtils.isBlank(tkXint)) {
+            try {
+                final URIBuilder uriBuilder = new URIBuilder(endpointUrl);
+                final List<NameValuePair> queryParams = uriBuilder.getQueryParams();
+                return uriBuilder.clearParameters()
+                        .setParameters(updatedQueryParameters(queryParams, tkXint))
+                        .build().toString();
+            } catch (URISyntaxException e) {
+                throw new PreBidException("Cant resolve endpoint parameters: " + endpointUrl, e);
+            }
+        } else {
+            return endpointUrl;
+        }
+    }
+
+    private List<NameValuePair> updatedQueryParameters(List<NameValuePair> queryParams, String tkXint) {
+        final List<NameValuePair> updatedParameters = queryParams.stream()
+                .filter(nameValuePair -> !nameValuePair.getName().equals(TK_XINT_QUERY_PARAMETER))
+                .collect(Collectors.toList());
+        updatedParameters.add(new BasicNameValuePair(TK_XINT_QUERY_PARAMETER, tkXint));
```

Comment:
Isn't it better performance-wise to just replace needed element in the list with `List.set(i,E)` (if the list implementation we get is mutable of course)?